test/Microsoft.AspNet.Mvc.Extensions.Test/JsonResultTest.cs

```diff
@@ -109,15 +112,23 @@ public class JsonResultTest
             Assert.Equal("application/json; charset=utf-8", context.Response.ContentType);
         }
 
-        private HttpContext GetHttpContext()
+        private static HttpContext GetHttpContext()
         {
             var httpContext = new DefaultHttpContext();
             httpContext.Response.Body = new MemoryStream();
 
+            var mockJsonOptions = new Mock<IOptions<MvcJsonOptions>>();
+            mockJsonOptions.Setup(m => m.Options).Returns(new MvcJsonOptions());
+
+            var serviceProvider = new Mock<IServiceProvider>();
+            serviceProvider.Setup(s => s.GetService(typeof(IOptions<MvcJsonOptions>)))
+                .Returns(mockJsonOptions.Object);
+            httpContext.RequestServices = serviceProvider.Object;
+
```

Comment:
This is also a case where it's pretty easy to avoid using Moq if you're so inclined. Not really important, but food for thought. You can create an `IServiceCollection` and call `AddOptions` and that would do what you need.